go/grpcweb/wrapper.go

```diff
@@ -9,88 +9,44 @@ import (
+	server *grpc.Server
+	opts   *options
 }
 
 // WrapServer takes a gRPC Server in Go and returns a WrappedGrpcServer that provides gRPC-Web Compatibility.
 //
 // The internal implementation fakes out a http.Request that carries standard gRPC, and performs the remapping inside
 // http.ResponseWriter, i.e. mostly the re-encoding of Trailers (that carry gRPC status).
-//
-// You can control the behaviour of the wrapper (e.g. modifying CORS behaviour) using `With*` options.
 func WrapServer(server *grpc.Server, options ...Option) *WrappedGrpcServer {
 	opts := evaluateOptions(options)
-	corsWrapper := cors.New(cors.Options{
-		AllowOriginFunc:  opts.originFunc,
-		AllowedHeaders:   append(opts.allowedRequestHeaders, internalRequestHeadersWhitelist...),
-		ExposedHeaders:   nil,                                 // make sure that this is *nil*, otherwise the WebResponse overwrite will not work.
-		AllowCredentials: true,                                // always allow credentials, otherwise :authorization headers won't work
-		MaxAge:           int(10 * time.Minute / time.Second), // make sure pre-flights don't happen too often (every 5s for Chromium :( )
-	})
-	websocketOriginFunc := opts.websocketOriginFunc
-	if websocketOriginFunc == nil {
-		websocketOriginFunc = defaultWebsocketOriginFunc
-	}
 	return &WrappedGrpcServer{
-		server:              server,
-		opts:                opts,
-		corsWrapper:         corsWrapper,
-		originFunc:          opts.originFunc,
-		enableWebsockets:    opts.enableWebsockets,
-		websocketOriginFunc: websocketOriginFunc,
+		server: server,
+		opts:   opts,
 	}
 }
 
 // ServeHTTP takes a HTTP request and if it is a gRPC-Web request wraps it with a compatibility layer to transform it to
 // a standard gRPC request for the wrapped gRPC server and transforms the response to comply with the gRPC-Web protocol.
-//
-// The gRPC-Web compatibility is only invoked if the request is a gRPC-Web request as determined by IsGrpcWebRequest or
-// the request is a pre-flight (CORS) request as determined by IsAcceptableGrpcCorsRequest.
-//
-// You can control the CORS behaviour using `With*` options in the WrapServer function.
 func (w *WrappedGrpcServer) ServeHTTP(resp http.ResponseWriter, req *http.Request) {
-	if w.enableWebsockets && w.IsGrpcWebSocketRequest(req) {
-		if w.websocketOriginFunc(req) {
-			if !w.opts.corsForRegisteredEndpointsOnly || w.isRequestForRegisteredEndpoint(req) {
-				w.HandleGrpcWebsocketRequest(resp, req)
-				return
-			}
-		}
-		resp.WriteHeader(403)
-		resp.Write(make([]byte, 0))
-		return
-	}
-
-	if w.IsAcceptableGrpcCorsRequest(req) || w.IsGrpcWebRequest(req) {
-		w.corsWrapper.Handler(http.HandlerFunc(w.HandleGrpcWebRequest)).ServeHTTP(resp, req)
-		return
+	if w.opts.enableWebsockets && w.IsGrpcWebSocketRequest(req) {
+		w.HandleGrpcWebsocketRequest(resp, req)
+	} else if w.IsGrpcWebRequest(req) {
+		w.HandleGrpcWebRequest(resp, req)
+	} else {
+		w.server.ServeHTTP(resp, req)
```

Comment:
If elses like this is usually considered a bit unidiomatic, instead early returns are encouraged to reduce indentation, like this:
```go
	if w.opts.enableWebsockets && w.IsGrpcWebSocketRequest(req) {
		w.HandleGrpcWebsocketRequest(resp, req)
		return
	}
	if w.IsGrpcWebRequest(req) {
		w.HandleGrpcWebRequest(resp, req)
		return
	} 
	w.server.ServeHTTP(resp, req)
```

What do you think?